File: monitor-data/src/main/java/africa/atps/monitordata/models/Scenario.java
package africa.atps.monitordata.models;

import dto.ENUM.Criticite;
import dto.ENUM.Tache;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Scenario implements Serializable {
    // TODO: 1/29/20 definir la classe scenario
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column
    private Criticite criticite;

    @Enumerated(EnumType.STRING)
    @Column
    @ElementCollection(targetClass = Tache.class)
    private Set<Tache> taches;

    @OneToOne
    private ServerHote serverHote;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "scenario")
    private List<Resultat> resultats;
}

File: monitor-view/src/app/shared/entities/Scenario.ts
import {IServerHote, ServerHote} from './ServerHote';
import {TacheEnum} from './TacheEnum';
import {IResultat, Resultat} from './IResultat';
import {CriticiteEnum} from './CriticiteEnum';

/**
 * Cette classe permet de africa.atps.monitordata.mapper les scenarios
 */
export interface IScenario {
  id?: number;
  criticite?: CriticiteEnum;
  serverHote?: IServerHote;
  taches?: TacheEnum[];
  resultat?: IResultat;
}

export class Scenario implements IScenario {
  id: number;
  criticite: CriticiteEnum;
  serverHote: ServerHote;
  taches: TacheEnum[];
  resultat: Resultat;
}
